### ISSUE TITLE
MD5 hash on the resource

### ISSUE THREAD

**Issue by metaodi** [davidread/ckanext-xloader#33]
I just found out, that xloader creates MD5 hashes for all files to check if a file is already in the DataStore. Those hashes are then saved on the resource.

This is perfectly fine, but we already generated a hash for each resource using SHA1 for our own use case (update a custom field on the dataset, if a resource has changed). So together with xloader, this created the situation, that every file has suddenly been seen as _changed_, since the MD5 hash did not match our SHA1 hash.

For now we solved this by simply switching to MD5 (https://github.com/opendatazurich/ckanext-stadtzh-harvest/pull/35). But maybe it might be worth having a config option to specify the algorithm used for the hashing or simply providing an interface, so plugins can actually change the way the hash is calculated.

At the very least, the current implementation needs a note in the README, so explain what's going on.

WDYT?

**Comment by metaodi** [comment]
Related to this issue I found a problem regarding our "conflicting" implementation of the hash. The way it currently works is, that our custom harvester processes a dataset and it's resources, calculates the hash (now as MD5) and set it on a resource. After saving all that, the xloader gets notified, checks the hash. But at this point, the hash on the ressource and the calculated one from the file will always match, as this is exactly what our harvester does.

To get around this problem, I submitted PR #34 to globally set the `ignore_hash` parameter.
